payment_gateway_adyen/services/payment_service.py

```diff
@@ -0,0 +1,173 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Akretion (http://www.akretion.com).
+# @author Raphael Valyi <raphael.valyi@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo.exceptions import Warning as UserError
+from odoo.tools.translate import _
+from odoo.tools.float_utils import float_round
+from odoo.addons.component.core import Component
+import json
+import logging
+_logger = logging.getLogger(__name__)
+
+try:
+    import Adyen
+except ImportError:
+    _logger.debug('Cannot import Adyen')
+
+
+MAP_SOURCE_STATE = {  # TODO check if complete
+    '[cancel-received]': 'cancel',
+    'Authorised': 'to_capture',
+    'consumed': 'succeeded',
+    'Refused': 'failed',
+    'RedirectShopper': 'pending',
+    '[capture-received]': 'succeeded'}
+
+# decimal points of currencies https://docs.adyen.com/developers/currency-codes
+ZERO_DECIMAL_CURRENCIES = [
+    u'CVE', u'DJF', u'GNF', u'IDR', u'JPY', u'KMF', u'KRW', u'PYG',
+    u'RWF', u'UGX', u'VND', u'VUV', u'XAF', u'XOF', u'XPF',
+],
+
+THREE_DECIMAL_CURRENCIES = [
+    u'BHD', u'JOD', u'KWD', u'LYD', u'OMR', u'TND'
+]
+
+
+class PaymentService(Component):
+    _inherit = 'payment.service'
+    _name = 'payment.service.adyen'
+    _usage = 'gateway.provider'
+    _allowed_capture_method = ['immediately']
+
+    def process_return(self, **params):
+        payload = {}
+        payload["browserInfo"] = {
+            "userAgent": "Mozilla/5.0 (X11; U; Linux i686; en-US; rv:\
```

Comment:
needed? Maybe we should used the header from the user browser

Reply:
yes this is required otherwise we get the following error from Adyen: ```'Provide the required request parameters to complete this request: \nbrowserInfo'```
As validation seems to happens with this fake userAgent and acceptHeader it is seems we don't need to pass the real data. But in doubt let's forward the true data as you suggested, I will propose this in the next commit.